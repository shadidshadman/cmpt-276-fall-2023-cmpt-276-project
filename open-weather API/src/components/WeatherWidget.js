import React, { useEffect } from 'react';

const WeatherWidget = () => {
  useEffect(() => {
    const script = document.createElement('script');
    script.src = "//openweathermap.org/themes/openweathermap/assets/vendor/owm/js/d3.min.js";
    document.body.appendChild(script);

    window.myWidgetParam = [{
      id: 11,
      cityid: '6173331', // City ID for Vancouver
      appid: '463bce9969af3f76b2c3f7ddecd37e80',
      units: 'metric',
      containerid: 'openweathermap-widget-11',
    }];

    const widgetScript = document.createElement('script');
    widgetScript.async = true;
    widgetScript.charset = "utf-8";
    widgetScript.src = "//openweathermap.org/themes/openweathermap/assets/vendor/owm/js/weather-widget-generator.js";
    document.body.appendChild(widgetScript);
  }, []);

  return (
    <>
      <h2>Sample Widget for Vancouver</h2>
      <div id="openweathermap-widget-11" />
    </>
  );
};

export default WeatherWidget;