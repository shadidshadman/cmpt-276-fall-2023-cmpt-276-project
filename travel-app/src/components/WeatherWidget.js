import React, { useEffect } from 'react';

const WeatherWidget = ({ cityId }) => {
  useEffect(() => {
    if (!window.myWidgetParam) {
      window.myWidgetParam = [];
    }

    window.myWidgetParam.push({
      id: 15,
      cityid: cityId,
      appid: '463bce9969af3f76b2c3f7ddecd37e80',
      units: 'metric',
      containerid: `openweathermap-widget-15-${cityId}`,
    });

    const script = document.createElement('script');
    script.async = true;
    script.charset = "utf-8";
    script.src = "//openweathermap.org/themes/openweathermap/assets/vendor/owm/js/weather-widget-generator.js";
    document.body.appendChild(script);
  }, [cityId]);

  return (
    <div id={`openweathermap-widget-15-${cityId}`} />
  );
};

export default WeatherWidget;