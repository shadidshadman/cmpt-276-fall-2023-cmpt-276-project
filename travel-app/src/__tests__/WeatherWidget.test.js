import { render } from '@testing-library/react';
import WeatherWidget from '../components/WeatherWidget.js';

test('WeatherWidget renders correctly with a cityId', () => {
  const { container } = render(<WeatherWidget cityId={123} />);
  expect(container).toMatchSnapshot();
});