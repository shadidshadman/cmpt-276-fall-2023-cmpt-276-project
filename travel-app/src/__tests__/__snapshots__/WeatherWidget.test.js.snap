// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidget renders correctly with a cityId 1`] = `
<div>
  <div
    id="openweathermap-widget-15-123"
  />
</div>
`;
